operator/go.mod

```diff
@@ -0,0 +1,79 @@
+module github.com/openshift/coredns-ocp-dnsnameresolver/operator
+
+go 1.19
```

Comment:
As the DNSNameResolver controller will be imported in cluster dns operator which uses `go 1.19` I have used that version for the operator. I'll update the README.md file.

Regarding the CI jobs, we can still use image with golang version 1.21 for running the jobs as it'll work for go 1.19 as well.

Reply:
> As the DNSNameResolver controller will be imported in cluster dns operator which uses go 1.19 I have used that version for the operator.

Did you see any compile errors related to this? I don't think the controller uses any 1.21 constructions since it builds green with `go 1.19`.

Reply:
There were no compile errors related to this.